File: app/src/main/java/com/parabola/newtone/mvp/view/RecentlyAddedPlaylistView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;

@AddToEndSingle
public interface RecentlyAddedPlaylistView extends MvpView {

    void refreshTracks(List<Track> trackList);
    void setItemViewSettings(TrackItemView viewSettings);
    void setItemDividerShowing(boolean showed);

    void removeTrack(int trackId);

    void setCurrentTrack(int trackId);

    @OneExecution
    void showToast(String message);

}

File: app/src/main/java/com/parabola/newtone/presentation/playlist/recentlyadded/RecentlyAddedPlaylistView.kt
package com.parabola.newtone.presentation.playlist.recentlyadded

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface RecentlyAddedPlaylistView : MvpView {
    fun setCurrentTrack(trackId: Int)
    fun refreshTracks(tracks: List<Track>)

    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)

    fun removeTrack(trackId: Int)
}
